logos-codegen/src/mir.rs

```diff
@@ -71,8 +68,11 @@ impl Mir {
             Mir::Empty | Mir::Loop(_) | Mir::Maybe(_) => 0,
             Mir::Concat(concat) => concat.iter().map(Mir::priority).sum(),
             Mir::Alternation(alt) => alt.iter().map(Mir::priority).min().unwrap_or(0),
-            Mir::Class(_) => 1,
-            Mir::Literal(_) => 2,
+            Mir::Class(_) => 2,
```

Comment:
Maybe we need to adjust `Class` to something different.
There is the `iter` function on [`ClassUnicode`](https://docs.rs/regex-syntax/latest/regex_syntax/hir/struct.ClassUnicode.html#method.iter) and those items have a [`len` method](https://docs.rs/regex-syntax/latest/regex_syntax/hir/struct.ClassUnicodeRange.html#method.len).
Maybe that is what we want instead of a hard coded class?! I'm not really sure tbh.

Reply:
I don't think it will have a different result: a non-empty class means that at it must match one element from the class. All "special" characters are escaped in a class, and you can only match one byte or one char → priority is 2